### PR TITLE
V2 alpha federated store alpha4 pr pt4

### PR DESCRIPTION
**Pull request by GCHQDev404**

### CODE REVIEW

File: store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/operation/handler/impl/FederatedAddGraphHandlerTest.java [hunk 1]
```diff
@@ -98,7 +91,7 @@ public void shouldAddGraph() throws Exception {
         store.initialise(FEDERATEDSTORE_GRAPH_ID, null, federatedStoreProperties);
         final Schema expectedSchema = new Schema.Builder().build();
 
-        assertThat(store.getGraphs(testUser, null, ignore)).hasSize(0);
+        assertThat(store.getGraphs(testUser, null, new AddGraph())).hasSize(0);
```
Comment (t92549) [inline store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/operation/handler/impl/FederatedAddGraphHandlerTest.java:94]: Definitely missing some context, but why did this change from `IgnoreOptions` to `AddGraph`?

Comment (GCHQDev404) [reply store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/operation/handler/impl/FederatedAddGraphHandlerTest.java:94]: IgnoreOptions was a TestClass created I think which did just that.
Now options is storing Federation flags its required.

File: store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/IFederatedOperation.java [hunk 2]
```diff
@@ -32,7 +32,7 @@
 public interface IFederatedOperation extends Operation {
 
     @JsonProperty("graphIds")
-    IFederatedOperation graphIdsCSV(final String graphIds);
+    IFederatedOperation graphIdsCSV(final String graphIds); //TODO FS request for this to be list.
```
[the remark this answers is not in the document]
Comment (GCHQDev404) [reply store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/IFederatedOperation.java:35]: I don't think I would want to have conflicting fields within an operation.
`String graphIdCSV`
`List<String> graphIds`
I don't think its intuitive which one is active or supersedes at a glance.

I'd rather have one field and different ways to set it. This is easy with a Builder but will require more attention with JSON creators.

[the remark this answers is not in the document]
Comment (t92549) [reply store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/IFederatedOperation.java:35]: That's fair, I just worry about breaking api compatibility for no gain to the user

**Pull request merged**